Bugfix for bug preventing the plotting of trajectories ending on exactly 0.00 longitude

pirmink opened the pull request:
Bugfix for bug preventing the plotting of trajectories when one ends with 0.00 longitude and next starts with abs(lon) > 20 as this was wrongly interpreted as crossing the date line.

Additionally, the long lines in the README.md are wrapped and some other minor edits are made. The patch level of the version is incremented by one. The license is changed to BSD 3-Clause License (the default for public MeteoSwiss projects). A minor bugfix has been made in the makefile (formerly Makefile). As this might be removed in a future version, use the --dry-run option to make to see what you would have to do manually without it.

pirmink committed:
Bugfix for trajectory data with longitude sign flip between trajectory and ancillary trajectory or between two ancillary trajectories.

pirmink committed:
Minor changes to README and changed licence to BSD 3-clause.

pirmink committed:
Added test for bugfix in version 1.1.2.

cosunae commented on src/pytrajplot/parse_data.py:
```diff
@@ -491,6 +491,11 @@ def check_input_dir(input_dir, prefix_dict, separator):
+    # check if plot_info has been found
+    if not "plot_info_dict" in dir():
+        print("ERROR: plot_info file not found:", prefix_dict["plot_info"])
+        raise
```
if no expression is present in the raise, it will re-raise the last exception raised: 
https://docs.python.org/2/reference/simple_stmts.html?highlight=raise#raise
This is useful for catch blocks, if you want to do something but letting also the exception by caught by others later in the stack. 
But in this context there is no exception being triggered, isnt? So it will most likely generate a TypeError, which seems to me not the intention. 
I would replace by 
`raise RuntimeError(".....") `
or equivalent exception and remove the print.

cosunae replied:
I would also recommend adding a test that triggers and captures the exception, to make sure the logic and behaviour is correct

pirmink replied:
Fixed.

cosunae commented on src/pytrajplot/plotting/analyse_trajectories.py:
```diff
@@ -82,6 +82,7 @@ def _get_traj_dict(data, number_of_trajectories=None, traj_length=None) -> dict:
 
 def _check_dateline_crossing(
     lon: pd.Series,
+    number_of_times: int,
```
can you document the new parameter?

cosunae replied:
notice also the doc of the return type is wrong.

pirmink replied:
Done.

cosunae commented on src/pytrajplot/plotting/analyse_trajectories.py:
```diff
@@ -108,7 +109,11 @@ def _check_dateline_crossing(
+            if (
+                not np.isnan(lon[flip_index])
+                and not np.isnan(lon[flip_index - 1])
+                and flip_index % number_of_times != 0
```
I guess this is the core of the bug fix, and I am lacking knowhow to properly review. 
I find `number_of_times` a confusing name that does not help understand the logic. From a different part of the code I read `#lines that make up one trajectory`. So would it be more appropriate to say `number_of_segments` ? 
Is a trajectory composed by a number of segments with all same number of points and the condition does want to ignore when moving between segments? 
If so, I would probably solve the problem differently by using a different data structure, a sequence of sequences for lon. So that the algorithms applied to only the inner most sequence can not mixed accross contiguous segments. That eliminates the need of this extra condition, and makes the intention more readable. 
But I am not sure I understood it

pirmink replied:
number_of_times is what is really meant, as a trajectory is composed of a (fixed per file) number of time steps, however restructuring the data as proposed could be beneficial --> created an issue for later

cosunae commented:
In the review we should check that all the checks are passing, precommit and all the tests. 
I would create a different PR just adding the precommit github workflows (see blueprint)
https://github.com/MeteoSwiss-APN/mch-python-blueprint/tree/main/tmpl/.github/workflows
and a jenkinsfile to setup a new jenkins plan (I assume there is not yet).
I would then test this PR against the newly added checks.

pirmink committed:
Documented variable, corrected error handling.

pirmink committed:
Corrected variable documentation, moved requirement files.

pirmink committed:
Merge branch 'main' into dateline-bug

pirmink committed:
End-of-file fixed.

pirmink committed:
Added dependabot configuration to have it update github actions automatically.

pirmink committed:
Minor fix in documentation.

pirmink committed:
Temporarily changing dependabot interval to daily to make it uptdate the actions.

pirmink committed:
Test dependabot.yml for github actions.

pirmink committed:
Removed non-functional dependabot.yml to update github actions.

pirmink committed:
Update pre-commit.yml

Updated versions of actions to avoid "Node.js 12 actions are deprecated" warnings.

pirmink committed:
Update pre-commit.yml

Removed unnecessarily detailed versions.

pirmink committed:
Adapted many files to new blueprint.
